atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransactionManagerImpl.java

```diff
@@ -249,27 +241,82 @@ public SerializableTransactionManagerImpl(KeyValueService keyValueService,
 
         this.keyValueService = keyValueService;
         this.timelockService = timelockService;
-        this.lockService = lockService;
-        this.transactionService = transactionService;
         this.conflictDetectionManager = conflictDetectionManager;
+        this.lockService = lockService;
         this.sweepStrategyManager = sweepStrategyManager;
-        this.constraintModeSupplier = constraintModeSupplier;
         this.cleaner = cleaner;
-        this.allowHiddenTableAccess = allowHiddenTableAccess;
-        this.lockAcquireTimeoutMs = lockAcquireTimeoutMs;
         this.closingCallbacks = new CopyOnWriteArrayList<>();
         this.isClosed = new AtomicBoolean(false);
         this.getRangesExecutor = createGetRangesExecutor(concurrentGetRangesThreadPoolSize);
         this.timestampTracker = timestampTracker;
-        this.defaultGetRangesConcurrency = defaultGetRangesConcurrency;
-        this.sweepQueueWriter = sweepQueueWriter;
+
+        transactionsFactory = new TransactionFactory(keyValueService,
+                timelockService,
+                transactionService,
+                cleaner,
+                conflictDetectionManager,
+                sweepStrategyManager,
+                constraintModeSupplier,
+                allowHiddenTableAccess,
+                timestampValidationReadCache,
+                lockAcquireTimeoutMs,
+                getRangesExecutor,
+                defaultGetRangesConcurrency,
+                sweepQueueWriter);
     }
 
     @Override
     protected boolean shouldStopRetrying(int numTimesFailed) {
         return numTimesFailed > NUM_RETRIES;
     }
 
+    @Override
+    public <C extends PreCommitCondition, E extends Exception> ServiceWriteTransaction getWriteTransaction(
+            C condition) throws E {
+        checkOpen();
+        try {
+            RawTransaction tx = setupRunTaskWithConditionThrowOnConflict(condition);
```

Comment:
If we can move all internal uses over to the transaction factory, we should see if we can drop setup/finishRunTask methods from the public API.